update-yarn/action.yml

```diff
@@ -14,8 +14,70 @@ runs:
           run: corepack enable
           shell: bash
 
-        - name: Update Yarn to the latest version
+        - name: Check if PR exists
+          id: check-pr-exists
+          shell: bash
           env:
               GH_TOKEN: ${{ github.token }}
+          run: |
+              LABEL_NAME="yarn-update"
+
+              # Create label if it doesn't exist
+              if ! gh label list | grep --quiet "$LABEL_NAME"; then
+                gh label create "$LABEL_NAME" --color "ededed" --description "A pull request created to update the Yarn package manager version."
+              fi
+
+              # Ensure pull request doesn't exist
+              EXISTING_YARN_PULL_REQUESTS=$(gh pr list --label "$LABEL_NAME" --state open)
+              if [ -z "$EXISTING_YARN_PULL_REQUESTS" ]; then
+                echo "No existing Yarn update PR found."
+                echo "UPDATE_PR_ALREADY_EXISTS=false" >> "$GITHUB_OUTPUT"
+              else
+                echo "Existing Yarn update PR found. Skipping update."
+                echo "UPDATE_PR_ALREADY_EXISTS=true" >> "$GITHUB_OUTPUT"
+                exit 0;
+              fi
+
+        - name: Update Yarn to the latest version
+          if: ${{ success() && steps.check-pr-exists.outputs.UPDATE_PR_ALREADY_EXISTS == 'false' }}
+          id: update-yarn
           run: bash ${{ github.action_path }}/update-yarn.sh
           shell: bash
+
+        - name: Create branch
+          if: ${{ success() && steps.update-yarn.outputs.NEW_YARN_VERSION != '' }}
+          shell: bash
+          env:
+              NEW_YARN_VERSION: ${{ steps.update-yarn.outputs.NEW_YARN_VERSION }}
+          run: |
+              RED='\033[0;31m'
+              BRANCH_NAME="yarn-update-$NEW_YARN_VERSION"
+
+              if git ls-remote --heads origin "$BRANCH_NAME" | grep --quiet "$BRANCH_NAME"; then
+                echo -e "${RED}Branch '$BRANCH_NAME' already exists.${RESET}" >&2
+                exit 1
+              fi
+
+              git checkout -b "$BRANCH_NAME" || exit 1
+              git push --set-upstream origin "$BRANCH_NAME"
+
+              echo "BRANCH_EXISTS=false" >> "$GITHUB_OUTPUT"
+
+        - name: Commit and push changes
+          if: ${{ success() && steps.update-yarn.outputs.NEW_YARN_VERSION != '' }}
+          uses: equinor/farfetched-actions/signed-commit-and-push@v2.8
```

Comment:
Hmm, I wonder if this and the dependabot update will end up in an infinite loop? 🤔

Reply:
I guess it will be alright as long as we're not auto-releasing. It will, however, always be one version behind I guess?

Reply:
Yes, I think it will. Here's my reasoning: Dependabot will see that we have v2.9 and create a PR to update this to v2.9. Say we see this and create v2.10, then 2.10 exists and Dependabot will create a new PR that we merge.

Why not point this to main instead of a specific version? That way we'll also avoid introducing Dependabot. When a new release is made, it won't be obvious that this action can be affected by a change in signed-commit-and-push. But then again, as a consumer of that action, we shouldn't create a release with a broken action, so I think it's safe enough.